lib/matplotlib/projections/polar.py

```diff
@@ -284,6 +426,181 @@ def view_limits(self, vmin, vmax):
         return mtransforms.nonsingular(min(0, vmin), vmax)
 
 
+class _ThetaShift(mtransforms.ScaledTranslation):
+    """
+    Apply a padding shift based on axes theta limits.
+
+    This is used to create padding for radial ticks.
+
+    Parameters
+    ----------
+    axes : matplotlib.axes.Axes
+        The owning axes; used to determine limits.
+    pad : float
+        The padding to apply, in points.
+    start : str, {'min', 'max', 'rlabel'}
+        Whether to shift away from the start (``'min'``) or the end (``'max'``)
+        of the axes, or using the rlabel position (``'rlabel'``).
+    """
+    def __init__(self, axes, pad, mode):
+        mtransforms.ScaledTranslation.__init__(self, pad, pad,
+                                               axes.figure.dpi_scale_trans)
+        self.set_children(axes._realViewLim)
+        self.axes = axes
+        self.mode = mode
+        self.pad = pad
+
+    def get_matrix(self):
+        if self._invalid:
+            if self.mode == 'rlabel':
+                angle = (
+                    np.deg2rad(self.axes.get_rlabel_position()) *
+                    self.axes.get_theta_direction() +
+                    self.axes.get_theta_offset()
+                )
+            else:
+                if self.mode == 'min':
+                    angle = self.axes._realViewLim.xmin
+                elif self.mode == 'max':
+                    angle = self.axes._realViewLim.xmax
+            angle %= 2 * np.pi
```

Comment:
It only looks like `angle` is used below in trig functions--why is all of the playing with cycles of `2 * pi` necessary? They should just work, regardless, right?

Reply:
I recall something weird happening before; it might have been when things were mixed degrees/radians and I hadn't realized which units were in use at this location. I will double-check to see if this is necessary.